node/src/neighborhood/mod.rs

```diff
@@ -3039,6 +3654,593 @@ mod tests {
+                CountryCodes {
+                    country_codes: vec!["CZ".to_string()],
+                    priority: 1,
+                },
+                CountryCodes {
+                    country_codes: vec!["FR".to_string()],
+                    priority: 2,
+                },
+            ],
+            show_countries: false,
+        };
+        let set_exit_location_message = NodeFromUiMessage {
+            client_id: 8765,
+            body: request.tmb(1234),
+        };
+        let mut subject = make_standard_subject();
+        let system = System::new(test_name);
+        let (ui_gateway, _, ui_gateway_recording_arc) = make_recorder();
+        subject.logger = Logger::new(test_name);
+        let cz = &mut make_node_record(3456, true);
+        cz.inner.country_code_opt = Some("CZ".to_string());
+        let r = &make_node_record(4567, true);
+        let fr = &mut make_node_record(5678, true);
+        fr.inner.country_code_opt = Some("FR".to_string());
+        let t = &make_node_record(7777, true);
+        let db = &mut subject.neighborhood_database.clone();
+        db.add_node(cz.clone()).unwrap();
+        db.add_node(t.clone()).unwrap();
+        db.add_node(r.clone()).unwrap();
+        db.add_node(fr.clone()).unwrap();
+        let mut dual_edge = |a: &NodeRecord, b: &NodeRecord| {
+            db.add_arbitrary_full_neighbor(a.public_key(), b.public_key());
+        };
+        dual_edge(&subject.neighborhood_database.root(), cz);
+        dual_edge(cz, t);
+        dual_edge(cz, r);
+        dual_edge(r, fr);
+        subject.neighborhood_database = db.clone();
+        let subject_addr = subject.start();
+        let peer_actors = peer_actors_builder().ui_gateway(ui_gateway).build();
+        let cz_public_key = cz.inner.public_key.clone();
+        let r_public_key = r.inner.public_key.clone();
+        let fr_public_key = fr.inner.public_key.clone();
+        let t_public_key = t.inner.public_key.clone();
+        let assert_country_undesirability_populated = AssertionsMessage {
+            assertions: Box::new(move |neighborhood: &mut Neighborhood| {
+                assert_eq!(
+                    neighborhood
+                        .neighborhood_database
+                        .node_by_key(&cz_public_key)
+                        .unwrap()
+                        .metadata
+                        .country_undesirability,
+                    0u32,
+                    "CZ - We expecting zero, country is with Priority: 1"
+                );
+                assert_eq!(
+                    neighborhood
+                        .neighborhood_database
+                        .node_by_key(&r_public_key)
+                        .unwrap()
+                        .metadata
+                        .country_undesirability,
+                    0u32,
+                    "We expecting 0, country is not considered for exit location, so country_undesirability doesn't matter"
```

Comment:
Tohle me bohuzel docela zarazi. 

Bojim se, ze by z toho mohly byt do budoucna problemy. 

Nevim, jakou podminkou to potom hasis, ale vypada to jako "hack" (en.). Vypada to, ze by bylo logictejsi priradit temto Nodum, ktere nejsou v "exit location" nejvyssi mozne cislo... i64::MAX. Problem je, ze by se muselo osetrit, aby to nekde nepreteklo na algebraicke operaci. 

Nicmene je hodne nesikovne, eufemisticky receno, ze jak ten Node s nejvyssi uzivatelovo prioritou ma stejnou hodnotu  jako Node, ktery je absolutne mimo hru. 
Neni to ciste resene...